cipher/diffiehellman.go

```diff
@@ -0,0 +1,21 @@
+package cipher
+
+import "github.com/TheAlgorithms/Go/math/modular"
+
+const (
+	generator         = 3
+	primeNumber int64 = 6700417 // prime number discovered by Leonhard Euler
+)
+
+// GenerateShareKey : generates a key using client private key , generator and primeNumber
+// this key can be made public
+// shareKey = (g^key)%primeNumber
+func GenerateShareKey(prvKey int64) (int64, error) {
+	return modular.Exponentiation(generator, prvKey, primeNumber)
+}
+
+// GenerateMutualKey : generates a mutual key that can be used by only alice and bob
```

Comment:
Probably add a wiki link. The reference to Alice and Bob may not be understood by everyone (inside jokes XD)